### ISSUE TITLE
Errors may have originated from an input operation. (FM.py)

### ISSUE THREAD

**Issue by elyorman** [cstur4/interaction-aware-factorization-machines#3]
`Errors may have originated from an input operation.
Input Source operations connected to node nonzero_embeddings:
 feature_embeddings/read (defined at FM.py:205)
 train_features_fm (defined at FM.py:90)

Original stack trace for 'nonzero_embeddings':
  File "FM.py", line 403, in <module>
    train(args)
  File "FM.py", line 330, in train
    model = FM(data.features_M, args.pretrain, make_save_file(args), args.hidden_factor, args.epoch, args.batch_size, args.lr, args.lamda, args.keep, args.optimizer, args.batch_norm, args.verbose, args.mla)
  File "FM.py", line 79, in __init__
    self._init_graph()
  File "FM.py", line 100, in _init_graph
    self.nonzero_embeddings = tf.nn.embedding_lookup(self.weights['feature_embeddings'], self.train_features, name='nonzero_embeddings')
  File "C:\Users\Admin\anaconda3\lib\site-packages\tensorflow\python\ops\embedding_ops.py", line 315, in embedding_lookup
    transform_fn=None)
  File "C:\Users\Admin\anaconda3\lib\site-packages\tensorflow\python\ops\embedding_ops.py", line 133, in _embedding_lookup_and_transform
    array_ops.gather(params[0], ids, name=name), ids, max_norm)
  File "C:\Users\Admin\anaconda3\lib\site-packages\tensorflow\python\util\dispatch.py", line 180, in wrapper
    return target(*args, **kwargs)
  File "C:\Users\Admin\anaconda3\lib\site-packages\tensorflow\python\ops\array_ops.py", line 3475, in gather
    return gen_array_ops.gather_v2(params, indices, axis, name=name)
  File "C:\Users\Admin\anaconda3\lib\site-packages\tensorflow\python\ops\gen_array_ops.py", line 4835, in gather_v2
    batch_dims=batch_dims, name=name)
  File "C:\Users\Admin\anaconda3\lib\site-packages\tensorflow\python\framework\op_def_library.py", line 788, in _apply_op_helper
    op_def=op_def)
  File "C:\Users\Admin\anaconda3\lib\site-packages\tensorflow\python\util\deprecation.py", line 507, in new_func
    return func(*args, **kwargs)
  File "C:\Users\Admin\anaconda3\lib\site-packages\tensorflow\python\framework\ops.py", line 3616, in create_op
    op_def=op_def)
  File "C:\Users\Admin\anaconda3\lib\site-packages\tensorflow\python\framework\ops.py", line 2005, in __init__
    self._traceback = tf_stack.extract_stack()`  

Can't run FM.py while INN.py and IFM.py both are running well. Also, the models are not being saved in the pretrain folder. Could you help me figure it out?

**Comment by cstur4** [comment]
It is may due to the memory issue.

**Comment by QuantumRaC** [comment]
running it with a cloud interpreter might help? can be caused by memory shortage
